### PR TITLE
Fix some typos in documentation

### PR DESCRIPTION
**Pull request by dylanpeifer**
This corrects a few simple typos I found in the documentation. There are a few other things that seemed like errors, but I wasn't sure. I'll list them here, but they are NOT fixed in this pull request.

1. In [finite fields](https://faculty.math.illinois.edu/Macaulay2/doc/Macaulay2/share/doc/Macaulay2/Macaulay2Doc/html/_finite_spfields.html) it says "Notice that the name of the generator is displayed with a $ in it", but this doesn't seem to be true in the example.

2. In [manipulating polynomials](https://faculty.math.illinois.edu/Macaulay2/doc/Macaulay2/share/doc/Macaulay2/Macaulay2Doc/html/_manipulating_sppolynomials.html) it says "Notice that the lead monomial is an element of a monoid whose name is [a,b]", but the monomial appears to be in R.

3. There are warnings from 2001 in [minimal primes of an ideal](https://faculty.math.illinois.edu/Macaulay2/doc/Macaulay2/share/doc/Macaulay2/Macaulay2Doc/html/_minimal_spprimes_spof_span_spideal.html) and [primary decomposition](https://faculty.math.illinois.edu/Macaulay2/doc/Macaulay2/share/doc/Macaulay2/Macaulay2Doc/html/_primary_spdecomposition.html). Are these still valid?

4. In [working with multiple rings](https://faculty.math.illinois.edu/Macaulay2/doc/Macaulay2/share/doc/Macaulay2/Macaulay2Doc/html/_working_spwith_spmultiple_springs.html) the example under "using maps to move between rings" doesn't seem to show what it claims to show, but maybe I'm just not understanding it.

5. The first example in [using functions](https://faculty.math.illinois.edu/Macaulay2/doc/Macaulay2/share/doc/Macaulay2/Macaulay2Doc/html/_using_spfunctions.html) does not actually show that sin takes a single argument.

6. In [inheritance](https://faculty.math.illinois.edu/Macaulay2/doc/Macaulay2/share/doc/Macaulay2/Macaulay2Doc/html/_inheritance.html) under the line starting "Now let's imagine we wish to treat" the example call of lookup doesn't return null as claimed. In this case lookup returns something from the file `robust.m2`.

I could edit these if given specific changes to make, but editing the source is probably just as fast as giving me changes to make here.

### CODE REVIEW

**Comment by DanGrayson** [comment]
That warning about lead monomials is long obsolete:
```

o9 = leadMonomial -- the leading monomial
     ************************************

     Description
     ===========

     leadMonomial f -- return the leading monomial of the polynomial or vector f, as a
     ring element.  (Warning: in version 0.9.2, a monoid element was returned.)

```


This warning seems too vague to have ever been true:
```
Warning (15 May 2001): If you stop a function mid process and then run minimalPrimes an error is given. Restarting Macaulay2 and then running minimalPrimes works around this.
```


@mikestillman, what about this warning?  Is still valid?
```
Warning (15 May 2001): This function is under construction. For example, the strategies, Monomial, GTZ and Hybrid are not written, or do not function as stated. Further, both the Monomial and Binomial strategies may be moved to separate functions.
```
In any case, it's too vague.  This is not the documentation node of a function, but it might be referring to `primaryDecomposition`.



The sentence "Since I is defined as an ideal in R2 we cannot type ZZ/101[s,t]/I as the computer sees ZZ/101[s,t] as different from R2 and so does not see I as being in this ring" is unclear because it is not explained that the problem is that each time you run ZZ/101[s,t] it creates a different ring.


This is strange: " this is shown by the fact that Macaulay2 sees them as -t^2 and 0 ", since M2 shows the name of the ring on the second output line.



That whole file `ov_ringmaps.m2` could be much improved by rewriting.  Much of it could be deleted or incorporated into a documentation for a specific function.


About "sin" taking one argument -- entering a function used to print documentation in a brief form, so maybe there is a good place to show what happens when you type "?sin".
```
i12 : ?sin

  * Usage: sin x
  * Inputs:
      * x, a real number
  * Outputs:
      * a real number, the sine of x

Ways to use sin
===============

  * sin(CC), see "sin" -- compute the sine
  * sin(QQ), see "sin" -- compute the sine
  * sin(RR), see "sin" -- compute the sine
  * sin(ZZ), see "sin" -- compute the sine
```


Ah, yes, the expression `lookup(symbol -, X) === null` is no longer true, because I installed a global method for printing an error message nicely.  Perhaps you could do what you think is appropriate.  If you want to demonstrate an error occurring when entering `- x` without having installPackage complain, just include `stopIfError = false` as part of the example code.

File: M2/Macaulay2/packages/Macaulay2Doc/ov_ringmaps.m2 [hunk 1]
```diff
@@ -186,10 +186,10 @@ document {
 	       	    "describe R4"
 	       	    },
      	       "The command ", TT "substitute", " works well to move elements
```
[the remark this answers is not in the document]
Comment (dylanpeifer) [reply M2/Macaulay2/packages/Macaulay2Doc/ov_ringmaps.m2]: Done in cc4069c139ce5806a474960824b932e57bf34c01.

File: M2/Macaulay2/packages/Macaulay2Doc/overview_modules.m2 [hunk 2]
```diff
@@ -499,7 +499,7 @@ document {
 	  "target G == M and source G == M",
 	  "inc * G == F"
 	  },
-     "Let's make sure that this map ", TT "G", " is welldefined.",
+     "Let's make sure that this map ", TT "G", " is well-defined.",
```
[the remark this answers is not in the document]
Comment (dylanpeifer) [reply M2/Macaulay2/packages/Macaulay2Doc/overview_modules.m2]: Done in cc4069c139ce5806a474960824b932e57bf34c01. I found both versions in various places throughout the documentation, and changed the others close by to be the same.

**Comment by dylanpeifer** [comment]
I used your suggestions to fix 2, 5, 6 and deleted the example in 1. Problems 3 and 4 are still there. I'm also getting a strange error when trying to make all the documentation:

    ./Macaulay2Doc/html/_changes_cm_sp1.11.html:27:13: error: broken link: ../../NonminimalComplexes/html/index.html

I'll try to track it down.

**Comment by DanGrayson** [comment]
Thanks!

**Pull request merged**